app/assets/javascripts/controllers/ems_common/ems_common_form_controller.js

```diff
@@ -298,5 +325,161 @@ ManageIQ.angular.app.controller('emsCommonFormController', ['$http', '$scope', '
+      if ($scope.newRecord) {
+        var metrics_password = $scope.emsCommonModel.metrics_password;
+        var metrics_verify = $scope.emsCommonModel.metrics_verify;
+      } else {
+        var metrics_password = $scope.emsCommonModel.metrics_password === "" ? "" : miqService.storedPasswordPlaceholder;
+        var metrics_verify = $scope.emsCommonModel.metrics_verify === "" ? "" : miqService.storedPasswordPlaceholder;
+      }
+      var metricsValidationModel = {
+        metrics_hostname:          $scope.emsCommonModel.metrics_hostname,
+        metrics_api_port:          $scope.emsCommonModel.metrics_api_port,
+        metrics_userid:            $scope.emsCommonModel.metrics_userid,
+        metrics_password:          metrics_password,
+        metrics_verify:            metrics_verify,
+      };
+      $scope.postValidationModel['metrics'] = metricsValidationModel;
+    } else if (prefix === "ssh_keypair") {
+      if ($scope.newRecord) {
+        var ssh_keypair_password = $scope.emsCommonModel.ssh_keypair_password;
+      } else {
+        var ssh_keypair_password = $scope.emsCommonModel.ssh_keypair_password === "" ? "" : miqService.storedPasswordPlaceholder;
+      }
+      $scope.postValidationModel['ssh_keypair'] = {
+        ssh_keypair_userid:        $scope.emsCommonModel.ssh_keypair_userid,
+        ssh_keypair_password:      ssh_keypair_password,
+      }
+    } else if (prefix === "service_account") {
+      $scope.postValidationModel['service_account'] = {
+        service_account:           $scope.emsCommonModel.service_account
+      }
+    } else if (prefix === "hawkular") {
+      $scope.postValidationModel['hawkular'] = {
+        hawkular_hostname:         $scope.emsCommonModel.hawkular_hostname,
+        hawkular_api_port:         $scope.emsCommonModel.hawkular_api_port,
+      }
+    }
+  };
+
+  $scope.validateClicked = function($event, authType, url, formSubmit) {
+    $scope.authType = authType;
+    miqService.validateWithREST($event, authType, url, formSubmit)
+      .then(function success(data) {
+        $scope.$apply(function() {
+          if(data.level == "error") {
+            if ($scope.authType === "default") {
+              $scope.emsCommonModel.default_auth_status = false;
+            } else if ($scope.authType === "amqp") {
+              $scope.emsCommonModel.amqp_auth_status = false;
+            } else if ($scope.authType === "service_account") {
+              $scope.emsCommonModel.service_account_auth_status = false;
+            } else if ($scope.authType === "metrics") {
+              $scope.emsCommonModel.metrics_auth_status = false;
+            } else if ($scope.authType === "ssh_keypair") {
+              $scope.emsCommonModel.ssh_keypair_auth_status = false;
+            } else if ($scope.authType === "hawkular") {
+              $scope.emsCommonModel.hawkular_auth_status = false;
+            }
+          } else {
+            if ($scope.authType === "default") {
```

Comment:
the else branch is exactly the same, except true instead of false .. maaybe ve could use a variable instread of copypasting?